pkg/broker/check_children_status.go

```diff
@@ -0,0 +1,167 @@
+package broker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Azure/open-service-broker-azure/pkg/async/model"
+	log "github.com/Sirupsen/logrus"
+)
+
+func (b *broker) doCheckChildrenStatuses(
+	ctx context.Context,
+	args map[string]string,
+) error {
+	instanceID, ok := args["instanceID"]
+	if !ok {
+		return errors.New(`missing required argument "instanceID"`)
+	}
+	instance, ok, err := b.store.GetInstance(instanceID)
+	if !ok {
+		return b.handleDeprovisioningError(
+			instanceID,
+			"checkChildrenStatus",
+			nil,
+			"error loading persisted instance",
+		)
+	}
+	if err != nil {
+		return b.handleDeprovisioningError(
+			instanceID,
+			"checkChildrenStatus",
+			err,
+			"error loading persisted instance",
+		)
+	}
+	childCount, err := b.store.GetInstanceChildCountByAlias(instance.Alias)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"step":       "checkChildrenStatus",
+			"instanceID": instanceID,
+			"error":      err,
+		}).Error(
+			"deprovisioning error: error determining child count",
+		)
+		return b.handleDeprovisioningError(
+			instance,
+			"checkChildrenStatus",
+			err,
+			"error determining child count",
+		)
+	}
+	var task model.Task
+	if childCount > 0 {
+		//Put this task back into the queue
+		task = model.NewDelayedTask(
+			"checkChildrenStatus",
+			map[string]string{
+				"instanceID": instanceID,
+			},
+			time.Minute*5,
+		)
+		log.WithFields(log.Fields{
+			"instanceID": instanceID,
+		}).Debug("children not deprovisioned, will wait again")
+	} else {
+		svc, ok := b.catalog.GetService(instance.ServiceID)
+		if !ok {
+			// If we don't find the Service in the catalog, something is really wrong.
+			// (It should exist, because an instance with this serviceID exists.)
+			log.WithFields(log.Fields{
+				"step":       "checkChildrenStatus",
+				"instanceID": instanceID,
+				"serviceID":  instance.ServiceID,
+			}).Error(
+				"deprovisioning error: no Service found for serviceID",
+			)
+			return b.handleDeprovisioningError(
+				instance,
+				"checkChildrenStatus",
+				nil,
+				"error: no Service found for serviceID",
+			)
+		}
+		plan, ok := svc.GetPlan(instance.PlanID)
+		if !ok {
+			// If we don't find the Service in the catalog, something is really wrong.
+			// (It should exist, because an instance with this serviceID exists.)
```

Comment:
This comment is wrong-- and I realize it was probably wrong where it was copied from too. Can we fix it in just this spot for now?